src/thread.cpp

```diff
@@ -80,6 +81,14 @@ void Thread::start_searching() {
   cv.notify_one(); // Wake up the thread in idle_loop()
 }
 
+void Thread::execute_with_worker(std::function<void(Thread&)> t)
```

Comment:
This code is a bit tricky because the ownership of `t` is moved to this thread, but `t` is not Rvalue.  I guess that this is because we want to copy `t` between multiple threads.  Could you please leave some comments about it?

Reply:
Ok, I clarified the ownership model of the worker tasks.

Reply:
The comments look good to me.  Thank you.